addons/mail/static/src/core/common/attachment_model.js

```diff
@@ -151,7 +151,7 @@ export class Attachment extends Record {
         const route = url(this.urlRoute, this.urlQueryParams);
         const encodedRoute = encodeURIComponent(route);
         if (this.isPdf) {
-            return `/web/static/lib/pdfjs/web/viewer.html?file=${encodedRoute}#pagemode=none`;
+            return url(`/web/static/lib/pdfjs/web/viewer.html?file=${encodedRoute}#pagemode=none`);
```

Comment:
If you are using the url helper here, you can take advantage of it and put `file` inside the `queryParams`.
You also don't need to `encodeURIComponent(route);` since the url helper will take care of it (through `queryParams` param).
If needed, the `pagemode=none` can just be concatenated at the end.